cli/azd/pkg/apphost/generate.go

```diff
@@ -611,17 +624,20 @@ func injectValueForBicepParameter(resourceName, p string, parameter any) (string
 		return finalParamValue, false, nil
 	}
 
-	if p == "keyVaultName" {
+	if p == knownParameterKeyVault {
 		return fmt.Sprintf("resources.outputs.SERVICE_BINDING_%s_NAME", strings.ToUpper(resourceName+"kv")), true, nil
 	}
-	if p == "principalId" {
-		return "resources.outputs.MANAGED_IDENTITY_PRINCIPAL_ID", true, nil
+	if p == knownParameterPrincipalId {
```

Comment:
Might consider storing these in a fixed map[string]string where we map the key to the value so we could just look up the value in the map and if it `has` an entry use it.  If we end up adding more types - maybe a reasonable refactoring.

Reply:
I have a little preference for constants in this case, to avoid loading the map into heap.